src/interaction/sessionrecording.cpp

```diff
@@ -223,25 +228,117 @@ bool SessionRecording::startRecording(const std::string& filename) {
 
         _timestampRecordStarted = global::windowDelegate->applicationTime();
 
-        //Record the current delta time so this is preserved in recording
-        double currentDeltaTime = global::timeManager->deltaTime();
-        std::string scriptCommandForInitializingDeltaTime =
-            "openspace.time.setDeltaTime(" + std::to_string(currentDeltaTime) + ")";
-        saveScriptKeyframe(scriptCommandForInitializingDeltaTime);
+        //Record the current delta time as the first property to save in the file.
+        //This needs to be saved as a baseline whether or not it changes during recording.
+        _timestamps3RecordStarted = {
+            _timestampRecordStarted,
+            0.0,
+            global::timeManager->time().j2000Seconds()
+        };
 
+        recordCurrentTimeRate();
         LINFO("Session recording started");
     }
 
     return recordingFileOK;
 }
 
+void SessionRecording::recordCurrentTimePauseState() {
+    bool isPaused = global::timeManager->isPaused();
+    std::string initialTimePausedCommand = "openspace.time.setPause(" +
+        std::string(isPaused ? "true" : "false") + ")";
+    saveScriptKeyframeToPropertiesBaseline(initialTimePausedCommand);
+}
+
+void SessionRecording::recordCurrentTimeRate() {
+    std::string initialTimeRateCommand = fmt::format(
+        "openspace.time.setDeltaTime({})", global::timeManager->deltaTime()
+    );
+    saveScriptKeyframeToPropertiesBaseline(initialTimeRateCommand);
+}
+
 void SessionRecording::stopRecording() {
     if (_state == SessionState::Recording) {
+        //Add all property baseline scripts to the beginning of the recording file
+        datamessagestructures::ScriptMessage smTmp;
+        for (timelineEntry initPropScripts : _keyframesSavePropertiesBaseline_timeline) {
+            if (initPropScripts.keyframeType == RecordedType::Script) {
+                smTmp._script = _keyframesSavePropertiesBaseline_scripts
+                    [initPropScripts.idxIntoKeyframeTypeArray];
+                saveSingleKeyframeScript(
+                    smTmp,
+                    _timestamps3RecordStarted,
+                    _recordingDataMode,
+                    _recordFile,
+                    _keyframeBuffer
+                );
+            }
+        }
+        for (timelineEntry entry : _timeline) {
+            switch (entry.keyframeType) {
+            case RecordedType::Camera:
```

Comment:
This is one of those cases (pun intended) where I've seen it done both ways. I agree that it's more readable indented, but this way seems to be more common (K&R style, java style, etc). I can't say I've been consistent with this because previously I wrote an indented case in the same file. I'm going to stick with this way here, particularly because it's already indented a lot and I won't have to wrap any lines :)